app/controllers/orders_controller.rb

```diff
@@ -0,0 +1,92 @@
+class OrdersController < ApplicationController
+  
+  before_action :authenticate_user!
+  before_action :set_journey, only: %i(new create checkout)
+  before_action :set_order, only: %i(checkout charge)
+  before_action :instantiate_order, only: %i(create)
+  before_action :ensure_booking_is_active, only: %i(new checkout create)
+  
+  def checkout
+  end
+  
+  def new
+    @minutes = Order.calculate_not_paid_minutes(@journey)
+    @order = Order.new
+    authorize(@order)
+  end 
+  
+  def create
+    @order.calculate_consumer_total_cents!
+    @order.state = Order::PENDING_PAYMENT
+    authorize(@order)
+    if @order.valid?
+      if @order.save
+        flash[:notice] = t 'orders.create.success'
+        redirect_to dashboard_path
+      else
+        flash[:alert] = @order.errors.full_messages
+        handle_errors
+      end
+    else
+      flash[:alert] = @order.errors.full_messages
+      handle_errors
+    end
+  end
+  
+  def charge
+    
+    raise "already charged" if @order.charged?
```

Comment:
Showing HTTP 500 errors to the user is never friendly

* redirect and `return` instead of raise
* Also report the issue to raygun and Rails.logger - as (I imagine) this line should never happen
* See `SafeLogger` (in my engine) for a sample class which does both things. You can implement a similar one.